signature/algorithm_test.go

```diff
@@ -0,0 +1,199 @@
+package signature
+
+import (
+	"crypto"
+	"crypto/ed25519"
+	"crypto/x509"
+	"reflect"
+	"testing"
+
+	"github.com/notaryproject/notation-core-go/testhelper"
+)
+
+func TestHash(t *testing.T) {
+	tests := []struct {
+		name   string
+		alg    Algorithm
+		expect crypto.Hash
+	}{
+		{
+			name:   "PS256",
+			alg:    AlgorithmPS256,
+			expect: crypto.SHA256,
+		},
+		{
+			name:   "ES256",
+			alg:    AlgorithmES256,
+			expect: crypto.SHA256,
+		},
+		{
+			name:   "PS384",
+			alg:    AlgorithmPS384,
+			expect: crypto.SHA384,
+		},
+		{
+			name:   "ES384",
+			alg:    AlgorithmES384,
+			expect: crypto.SHA384,
+		},
+		{
+			name:   "PS512",
+			alg:    AlgorithmPS512,
+			expect: crypto.SHA512,
+		},
+		{
+			name:   "ES512",
+			alg:    AlgorithmES512,
+			expect: crypto.SHA512,
+		},
+		{
+			name:   "UnsupportedAlgorithm",
+			alg:    0,
+			expect: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			hash := tt.alg.Hash()
+			if hash != tt.expect {
+				t.Fatalf("Expected %v, got %v", tt.expect, hash)
+			}
+		})
+	}
+}
+
+func TestExtractKeySpec(t *testing.T) {
+	tests := []struct {
+		name      string
+		cert      *x509.Certificate
+		expect    KeySpec
+		expectErr bool
+	}{
+		{
+			name: "RSA 3072",
+			cert: testhelper.GetRSALeafCertificate().Cert,
+			expect: KeySpec{
+				Type: KeyTypeRSA,
+				Size: 3072,
```

Comment:
For valid cases, can we enumerate all possible keySpec? eg: RSA 2048, RSA 4096
```suggestion
               // Same for ecdsa,  just an example
		for _, k := range []int{2048, 3072, 4096} {
			rsaRoot := testhelper.GetRSARootCertificate()
		        pk, _ := rsa.GenerateKey(rand.Reader, k)
		        certTuple := testhelper.GetRSACertTupleWithPK(pk, "TestWithRSA_"+strconv.Itoa(pk.Size()), &rsaRoot)
		        tests = append(tests, []struct{
		                name: "RSA"+strconv.Itoa(k),
		                cert: certTuple.Cert,
		                expect: KeySpec{
		                      Type: KeyTypeRSA,
		                      SIze: k,
		                },
		                expectErr: false,
		        }
		 }
```

Reply:
Good call, I'll add all valid cases.